### ISSUE TITLE
CoreInputView.OcclusionsChanged not providing occlusions

### ISSUE THREAD

**Issue by buhcsa** [microsoft/microsoft-ui-xaml#10210]
### Describe the bug

Unlike in version 1.5.x of Microsoft.WindowsAppSDK, the CoreInputView.OcclusionsChanged event in version 1.6.x does not provide occlusions in the argument CoreInputViewOcclusionsChangedEventArgs

### Steps to reproduce the bug

Create a new WinUI3 desktop app project, named "WinUI3App1".
Replace:
- MainWindow.xaml

```
<?xml version="1.0" encoding="utf-8"?>
<Window x:Class="WinUI3App1.MainWindow" xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml">
    <TextBox x:Name="myTextBox" HorizontalAlignment="Center" VerticalAlignment="Center"/>
</Window>
```

- MainWindow.cs

```
using Microsoft.UI.Xaml;
using System.Diagnostics;
using Windows.UI.ViewManagement.Core;

namespace WinUI3App1;

public sealed partial class MainWindow : Window
{
    readonly CoreInputView inputView;
    public MainWindow()
    {
        InitializeComponent();
        (inputView = CoreInputView.GetForCurrentView()).OcclusionsChanged += (s, e) =>
            Debug.WriteLine($"Occlusions.Count={e.Occlusions.Count}");
        myTextBox.GettingFocus += (s, e) =>
            inputView.TryShow(CoreInputViewKind.Keyboard);
        myTextBox.LosingFocus += (s, e) =>
            inputView.TryHide();
    }
}
```

Use a high screen resolution (e.g. 1920 x 1200) and run the app in debug mode.
If the text field is focused, the input pane is shown and the debug output “Occlusions.Count=0” is generated due to the occlusion by the input pane. No occlusions are provided.

### Expected behavior

The CoreInputView.OcclusionsChanged event provides occlusions in the CoreInputViewOcclusionsChangedEventArgs argument, if applicable.
In the example, if the text field is focused, the debug output “Occlusions.Count=1” is generated due to the occlusion by the input pane.

### NuGet package version

Windows App SDK 1.6.3: 1.6.241114003

### Packaging type

Unpackaged

### Windows version

Windows 11 version 22H2 (22621, 2022 Update)

### IDE

Visual Studio 2022

### Additional context

[WinUI3App1.zip](https://github.com/user-attachments/files/17939063/WinUI3App1.zip)

**Comment by buhcsa** [comment]
> This is a UWP API only.

Sorry, but it worked fine in WinUI3 desktop app / WinRT, _not_ UWP, with version 1.5.x of Microsoft.WindowsAppSDK. So it can't be a UWP API only.
See [https://learn.microsoft.com/en-us/uwp/api/windows.ui.viewmanagement.core.coreinputview.occlusionschanged?view=winrt-26100](url)
"Applies to WinRT, Build 16299, Build 17134, Build 17763, Build 18362, Build 19041, Build 20348, Build 22000, Build 22621, Build 26100".
Please review the issue again. Thx.

**Comment by buhcsa** [comment]
@RDMacLachlan @DarranRowe 

You have correctly stated based on [Windows Runtime APIs not supported in desktop apps](https://learn.microsoft.com/en-us/windows/apps/desktop/modernize/desktop-to-uwp-supported-api#classes-with-an-xxxforcurrentview-method):

> there is a note that states:
> 
> "CoreInputView.GetForCurrentView _is supported in desktop apps_, and it can be used even without a CoreWindow. You can use that method to retrieve a CoreInputView object on any thread; and if that thread has a foreground window, then that object will produce events."
> 
> So this is supported in any destkop applications according to official documentation.

CoreInputView.GetForCurrentView works correctly and the CoreInputView.OcclusionsChanged event is provided. The event is raised correctly.
The problem is simply that occlusions are not provided.

Why is an event being raised according to the documentation without the relevant information being included in the arguments? This leads to confusion and at least limits the potential of the API.
If an event is raised according to the API, it should also contain the information according to the API. Instead, it actually contains incorrect information.

In my view, the implementation of this API is incorrect in this respect and should be corrected. Please review the issue.

@DarranRowe
You are absolutely right with your side note. But in my opinion it does not apply to this specific point, as there is an official API and a different behavior is implemented than the one agreed there. So it is not simply observed behavior, but a contract.

**Comment by RDMacLachlan** [comment]
Hi @buhcsa,
I've requested follow-up from the development team on this GitHub Issue.

**Comment by buhcsa** [comment]
Hi @RDMacLachlan,
Thank you for your quick response.

**Comment by DarranRowe** [comment]
> You are absolutely right with your side note. But in my opinion it does not apply to this specific point, as there is an official API and a different behavior is implemented than the one agreed there. So it is not simply observed behavior, but a contract.

I disagree. Your original point was that CoreInputView.OcclusionsChanged must work in 1.6 because it worked in 1.5. This on its own was not a strong enough reason.

If you look back historically, there are plenty of undocumented behaviours that have changed that caused behaviour to work in one version of Windows, or with one version of a library but then caused crashes in the next versions. There are video drivers that modified the video control panel window in Windows 95, but this wasn't supported. There were shell extensions that relied on certain COM calls coming in a specified order, and this caused Explorer crashes for Windows Vista. Observing behaviour in one version of a library does not mean that it is supported and is expected to continue being supported.

This was precisely the reason why I provided the extra documentation stating that CoreInputView.OcclusionsChange was supported in the desktop environment. This changed this to an issue that was fully supported by Windows.

**Comment by DarranRowe** [comment]
> My actual point was that the implementation does not provide the officially described behavior of the API.
I realized this because a product I developed no longer behaves correctly with the new SDK version and customers have noticed this. This is the only reason why I referred to the older SDK versions.

Yes, and my point was that none of the information that you provided indicated that it **should** for desktop applications. Remember that WinUI 3 applications are not UWP applications, they are desktop applications. This unfortunately means that the CoreInputView.OcclusionsChanged working for WinUI 3 that comes with 1.5 can be seen as an outlier, and the behaviour with 1.6 follows the percieved behaviour documented for desktop applications.

The extra documentation that I linked to indicated that it **should** work with desktop applications too.

**Comment by buhcsa** [comment]
@DarranRowe 
That's right.
